traffic_ops/testing/api/v4/cachegroups_test.go

```diff
@@ -32,99 +35,393 @@ import (
 
 func TestCacheGroups(t *testing.T) {
 	WithObjs(t, []TCObj{Types, Parameters, CacheGroups, CDNs, Profiles, Statuses, Divisions, Regions, PhysLocations, Servers, Topologies}, func() {
-		ReadUpdateTestCacheGroupWithNullLatLong(t)
-		GetTestCacheGroupsIMS(t)
-		GetTestCacheGroupsByNameIMS(t)
-		GetTestCacheGroupsByShortNameIMS(t)
-		GetTestCacheGroups(t)
-		GetTestCacheGroupsByName(t)
-		GetTestCacheGroupsByShortName(t)
-		GetTestCacheGroupsByTopology(t)
-		CheckCacheGroupsAuthentication(t)
-		currentTime := time.Now().UTC().Add(-5 * time.Second)
-		time := currentTime.Format(time.RFC1123)
-		var header http.Header
-		header = make(map[string][]string)
-		header.Set(rfc.IfModifiedSince, time)
-		header.Set(rfc.IfUnmodifiedSince, time)
-		UpdateTestCacheGroups(t)
-		UpdateTestCacheGroupsWithHeaders(t, header)
-		GetTestCacheGroupsAfterChangeIMS(t, header)
-		header = make(map[string][]string)
-		etag := rfc.ETag(currentTime)
-		header.Set(rfc.IfMatch, etag)
-		UpdateTestCacheGroupsWithHeaders(t, header)
-		GetTestPaginationSupportCg(t)
-		GetTestCacheGroupsByInvalidId(t)
-		GetTestCacheGroupsByInvalidType(t)
-		GetTestCacheGroupsByType(t)
-		DeleteTestCacheGroupsByInvalidId(t)
-		UpdateCachegroupWithLocks(t)
+
+		methodTests := map[string]map[string]struct {
+			endpointId     func() int
+			clientSession  *client.Session
+			requestParams  map[string]string
+			requestBody    map[string]interface{}
+			requestHeaders http.Header
+			expectations   []utils.CkReqFunc
+		}{
+			"GET": {
+				"OK when VALID NAME parameter AND Lat/Long are 0": {
+					nil, TOSession, map[string]string{"name": "nullLatLongCG"}, nil, nil,
```

Comment:
This code would fail `go vet -composites` because it uses unkeyed fields. The rationale there being that if you add something or rearrange the structure of the type, the code will all fail to compile even if it still has exactly the information needed to work properly - because the fields will be improperly associated. We don't enforce that (yet, possibly, see #6545). That would also make it easier to read these without needing to scroll up to the type definition. Plus, with keyed fields we can rely on the zero values of unspecified properties, which allows us to omit  all these `nil`s I see.

Reply:
> We don't enforce that (yet, possibly, see #6545)

We do enforce it, but when 2 structs are on 1 line and one of them is keyed, `go vet` doesn't detect unkeyed fields in the second, which is a `go vet` limitation.